worlds/yachtdice/__init__.py

```diff
@@ -0,0 +1,459 @@
+import math
+
+from BaseClasses import CollectionState, Entrance, Item, Region, Tutorial
+
+from worlds.AutoWorld import WebWorld, World
+
+from .Items import YachtDiceItem, item_groups, item_table
+from .Locations import YachtDiceLocation, all_locations, ini_locations
+from .Options import YachtDiceOptions, yd_option_groups
+from .Rules import dice_simulation, set_yacht_completion_rules, set_yacht_rules
+
+
+class YachtDiceWeb(WebWorld):
+    tutorials = [
+        Tutorial(
+            "Multiworld Setup Guide",
+            "A guide to setting up Yacht Dice. This guide covers single-player, multiworld, and website.",
+            "English",
+            "setup_en.md",
+            "setup/en",
+            ["Spineraks"],
+        )
+    ]
+
+    option_groups = yd_option_groups
+
+
+class YachtDiceWorld(World):
+    """
+    Yacht Dice is a straightforward game, custom-made for Archipelago,
+    where you cast your dice to chart a course for high scores,
+    unlocking valuable treasures along the way.
+    Discover more dice, extra rolls, multipliers,
+    and unlockable categories to navigate the depths of the game.
+    Roll your way to victory by reaching the target score!
+    """
+
+    game: str = "Yacht Dice"
+    options_dataclass = YachtDiceOptions
+
+    web = YachtDiceWeb()
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+
+    location_name_to_id = {name: data.id for name, data in all_locations.items()}
+
+    item_name_groups = item_groups
+
+    ap_world_version = "2.0.4"
+
+    def _get_yachtdice_data(self):
+        return {
+            # "world_seed": self.multiworld.per_slot_randoms[self.player].getrandbits(32),
+            "seed_name": self.multiworld.seed_name,
+            "player_name": self.multiworld.get_player_name(self.player),
+            "player_id": self.player,
+            "race": self.multiworld.is_race,
+        }
+
+    # In generate early, we fill the item-pool, then determine the number of locations, and add filler items.
+    def generate_early(self):
+        self.itempool = []
+        self.precollected = []
+
+        # number of dice and rolls in the pull
+        ind_dice_rolls = self.options.minimal_number_of_dice_and_rolls.value
+
+        num_of_dice = [0, 2, 5, 5, 6, 7, 8][ind_dice_rolls]
+        num_of_rolls = [0, 2, 3, 5, 4, 3, 2][ind_dice_rolls]
+
+        # amount of dice and roll fragments needed to get a dice or roll
+        frags_per_dice = self.options.number_of_dice_fragments_per_dice.value
+        frags_per_roll = self.options.number_of_roll_fragments_per_roll.value
+
+        # count number of plando items not from pool, we need extra locations for them
+        self.extra_plando_items = 0
+
+        for plando_setting in self.multiworld.plando_items[self.player]:
+            if plando_setting.get("from_pool", False) is False:
+                self.extra_plando_items += sum(value for value in plando_setting["items"].values())
```

Comment:
Thanks for all the comments, all except this one are clear and I will start working on em!

The main culprit why I have this code is because someone added this to the yaml (with items plando in host.yaml on):
```
  plando_items:
    - items:
        Dice Fragment: 5
      from_pool: false
      world: true # I prefer those items to be placed in other worlds
      force: true
```
Just genning Yacht Dice with this yaml gives the error/warning
```
Unplaced items(5): [Encouragement (Player 1), Encouragement (Player 1), Encouragement (Player 1), Bonus Point (Player 1), Bonus Point (Player 1)] - Unfilled Locations(0): []
Per-Player counts: {'items': Counter({1: 85}), 'locations': Counter({1: 80})})
```
If I add 50 Dice Fragments instead, the gen fails because there is not enough locations for progression items.

I have asked in the discord for further feedback

Reply:
removed :D